### ISSUE TITLE
Maximum height allowed in the browser is too small

### ISSUE THREAD

**Issue by DougRzz** [pbugnion/ipywidgets_server#9]
Hi,  I am trying to break your code but I have failed! I am very impressed.  Basically, it can be used as an very effective dashboarding app for Jupyter. I have tested it with numerous widgets including several buttons, select widgets and Bqplot line & bar charts together with Vaex for large datasets  and I have had no real issues.  I have had to re-factor my Jupyter code a bit for it to work. This included removing any ipython.display commands. None of it a big deal.  

I have noticed that there seems to be a limit in the total height allowed in the webpage area.  If I zoom in the Chrome browser, the upper widgets disappear and the vertical scrollbar doesn't allow me to navigate to the upper display area.   Did you create a hard limit for the height?

**Comment by pbugnion** [comment]
That's really great to hear! Thanks for trying to break it -- it's very much appreciated. Do keep on posting bug reports.

> I have noticed that there seems to be a limit in the total height allowed in the webpage area.

My knowledge of CSS is somewhat inadequate, but I think there should be a height limit of 100% of the screen area. At the moment, it looks like widgets which are taller than this get truncated at the top, as you say. 

Would you agree that the correct behaviour for widgets taller than the screen height would be to be truncated at the bottom, and for a scrollbar to appear that lets the user see the rest of the widgets?

For now, a work-around to get something akin to this behaviour is to wrap the widget you want to display in something like:

```python
vbox = widgets.VBox([very_tall_widget], layout={'height': '500px', 'overflow_y': 'scroll'})
```

**Comment by DougRzz** [comment]
> Would you agree that the correct behaviour for widgets taller than the screen height would be to be truncated at the bottom, and for a scrollbar to appear that lets the user see the rest of the widgets?

Yes. Exactly

I've tried the layout settings and it kind of bunches up some of the widgets with several vertical scroll bars.  Its not really what I want.   I will play about with the CSS settings and see how far I get.

**Comment by pbugnion** [comment]
This should now be fixed.

I have been trying things out with the following snippet (among others):

```
import difflib
import random
import requests
import ipywidgets as widgets

CHECKBOX_KWARGS = {
    'value': False,
    'indent': False,
    'layout': {'display': 'block'}
}


def multi_checkbox_widget(descriptions):
    """ Widget with a search field and lots of checkboxes """
    search_widget = widgets.Text(layout={'display': 'block', 'width': '300px'})
    options_dict = {
        description: widgets.Checkbox(description=description, **CHECKBOX_KWARGS)
        for description in descriptions
    }
    options = [options_dict[description] for description in descriptions]
    options_widget = widgets.VBox(
        options,
        layout={'overflow_y': 'scroll'}
    )
    multi_select = widgets.VBox([search_widget, options_widget])

    # Wire the search field to the checkboxes
    def on_text_change(change):
        search_input = change['new']
        if search_input == '':
            # Reset search field
            new_options = [options_dict[description] for description in descriptions]
        else:
            # Filter by search field using difflib.
            close_matches = difflib.get_close_matches(search_input, descriptions, cutoff=0.0)
            new_options = [options_dict[description] for description in close_matches]
        options_widget.children = new_options

    search_widget.observe(on_text_change, names='value')
    return multi_select

# Example of using the widget

# Get lots of words for our options
words_url = 'https://svnweb.freebsd.org/csrg/share/dict/words?view=co&content-type=text/plain'
response = requests.get(words_url)
response.raise_for_status()
words = response.text
words = set([word.lower() for word in words.splitlines()])
descriptions = random.sample(words, 100)

w = multi_checkbox_widget(descriptions)
vbox = widgets.VBox(
    [w],
    layout={'height': '500px', 'overflow': 'scroll', 'align_items': 'center'}
)
```

Then for a tall widget:

```
ipywidgets-server test:w
```

or, for a shorter widget that manages its own centering and overflow:

```
ipywidgets-server test:vbox
```

The former looks like this:

<img width="584" alt="screen shot 2017-10-16 at 07 31 34" src="https://user-images.githubusercontent.com/1392879/31598181-1e03ab3a-b244-11e7-97f6-48fc19fb5c53.png">

The latter looks like this -- apart from managing its own vertical space, by setting `'align_items': 'center'` in the layout of the `vbox`, it makes the entire page horizontally centered.

<img width="1438" alt="screen shot 2017-10-16 at 07 32 49" src="https://user-images.githubusercontent.com/1392879/31598199-3fd10406-b244-11e7-8ccb-a3a786bc0fbb.png">

**Comment by pbugnion** [comment]
This is now available in ipywidgets_server 0.1.1.

```
$ pip install -U ipywidgets_server
```

**Comment by DougRzz** [comment]
This enhancement works really well. Thanks.
